python/lsst/ts/watcher/rules/mtm2_tangent_link_temperature.py

```diff
@@ -0,0 +1,164 @@
+# This file is part of ts_watcher.
+#
+# Developed for Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["MTM2TangentLinkTemperature"]
+
+import logging
+import types
+import typing
+
+import numpy as np
+import yaml
+from lsst.ts import salobj, watcher
+from lsst.ts.xml.enums.Watcher import AlarmSeverity
+
+from ..base_rule import AlarmSeverityReasonType, NoneNoReason
+from ..remote_info import RemoteInfo
+
+
+class MTM2TangentLinkTemperature(watcher.PollingRule):
```

Comment:
There is a BaseESSRule that does something very similar to what this rule does. Have you taken a look at it? I am not sure it handles the data format from M2 temperature but I would suggest you take a look at it.

Reply:
I checked this before:

https://ts-watcher.lsst.io/py-api/lsst.ts.watcher.BaseEssRule.html#lsst.ts.watcher.BaseEssRule

It looks like it was designed to compare with the fixed value threshold instead of a relative value from another telemetry topic.